src/main/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestController.java

```diff
@@ -0,0 +1,92 @@
+package edu.ucsb.cs156.rec.controllers;
+
+import edu.ucsb.cs156.rec.entities.RecommendationRequest;
+import edu.ucsb.cs156.rec.errors.EntityNotFoundException;
+import edu.ucsb.cs156.rec.models.CurrentUser;
+import edu.ucsb.cs156.rec.repositories.RecommendationRequestRepository;
+import io.swagger.v3.oas.annotations.Operation;
+import io.swagger.v3.oas.annotations.Parameter;
+import io.swagger.v3.oas.annotations.tags.Tag;
+
+import java.time.LocalDateTime;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@Tag(name = "RecommendationRequest")
+@RequestMapping("/api/recommendationrequest")
+@RestController
+public class RecommendationRequestController extends ApiController {
+    @Autowired
+    RecommendationRequestRepository recommendationRequestRepository;
+
+    /**
+     * This method returns a list of all recommendationrequests.
+     * @return a list of all recommendationrequests
+     */
+    @Operation(summary = "List all recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("/all")
+    public Iterable<RecommendationRequest> allRecommendationRequests() {
+        Iterable<RecommendationRequest> recommendationRequests = recommendationRequestRepository.findAll();
+        return recommendationRequests;
+    }
+
+    /**
+     * This method returns a single recommendationrequests.
+     * @param id id of the recommendationrequests to get
+     * @return a single recommendationrequests
+     */
+    @Operation(summary = "Get a single recommendation request")
+    @PreAuthorize("hasRole('ROLE_USER')")
+    @GetMapping("")
+    public RecommendationRequest getById(
+            @Parameter(name = "id") @RequestParam Long id) {
+            RecommendationRequest recommendationRequest = recommendationRequestRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(RecommendationRequest.class, id));
+
```

Comment:
Also need a check that this recommendation was requested by the current user.

There are at least two ways to do this check.

One way to do it is shown here; this does a search by id and by user at the same time; the search will return "not found" unless both fields match:

https://github.com/ucsb-cs156/proj-courses/blob/4bdc2674508428d03cb2dcd2e9077d75cdaf5a3f/src/main/java/edu/ucsb/cs156/courses/controllers/PersonalSchedulesController.java#L54

A different way is shown here—note this one first checks to see if the user is the author, and then also checks to see if the user is an admin.  We probably don't need the second check here, though it also probably wouldn't hurt anything to include it.

https://github.com/ucsb-cs156/proj-happycows/blob/7a5456e61fffafa73a3464eb16360c4aa06cd50f/src/main/java/edu/ucsb/cs156/happiercows/controllers/ChatMessageController.java#L157

Finally, we may need to consider that the user could be either the requester or the professor, and in *both* cases we should retrieve the record.  So some combination of the two may be appropriate.

What we must NOT do is allow users to look at each others recommendation requests. That's a major breach of privacy, and in this case, federal law (FERPA).